test/system/src/test/java/io/pravega/test/system/DynamicRestApiTest.java

```diff
@@ -0,0 +1,139 @@
+import io.pravega.client.stream.ScalingPolicy;
+import io.pravega.client.stream.StreamConfiguration;
+import io.pravega.client.stream.impl.DefaultCredentials;
+import lombok.Cleanup;
+import lombok.extern.slf4j.Slf4j;
+
+import io.pravega.test.system.framework.Environment;
+import io.pravega.test.system.framework.SystemTestRunner;
+import io.pravega.test.system.framework.Utils;
+import io.pravega.test.system.framework.services.Service;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.List;
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.ClientBuilder;
+import javax.ws.rs.client.Invocation;
+import javax.ws.rs.client.WebTarget;
+import javax.ws.rs.core.Response;
+
+import org.apache.commons.lang3.RandomStringUtils;
+import org.glassfish.jersey.client.authentication.HttpAuthenticationFeature;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.Timeout;
+import org.junit.runner.RunWith;
+import mesosphere.marathon.client.MarathonException;
+
+import static org.junit.Assert.assertEquals;
+import static javax.ws.rs.core.Response.Status.OK;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+@Slf4j
+@RunWith(SystemTestRunner.class)
+public class DynamicRestApiTest extends AbstractSystemTest {
+
+    @Rule
+    public Timeout globalTimeout = Timeout.seconds(5 * 60);
+
+    private final Client client;
+    private WebTarget webTarget;
+    private String restServerURI;
+    private String resourceURl;
+
+    public DynamicRestApiTest() {
+
+        org.glassfish.jersey.client.ClientConfig clientConfig = new org.glassfish.jersey.client.ClientConfig();
+        clientConfig.register(JacksonJsonProvider.class);
+        clientConfig.property("sun.net.http.allowRestrictedHeaders", "true");
+
+        if (Utils.AUTH_ENABLED) {
+            HttpAuthenticationFeature feature = HttpAuthenticationFeature.basic("admin", "1111_aaaa");
+            clientConfig.register(feature);
+        }
+
+        client = ClientBuilder.newClient(clientConfig);
+    }
+
+    /**
+     * This is used to setup the various services required by the system test framework.
+     *
+     * @throws InterruptedException If interrupted
+     * @throws MarathonException    when error in setup
+     * @throws URISyntaxException   If URI is invalid
+     */
+    @Environment
+    public static void initialize() throws MarathonException {
+        URI zkUri = startZookeeperInstance();
+        startBookkeeperInstances(zkUri);
+        URI controllerUri = ensureControllerRunning(zkUri);
+        ensureSegmentStoreRunning(zkUri, controllerUri);
+    }
+
+    @Test
+    public void listScopes() {
+        Service conService = Utils.createPravegaControllerService(null);
+        List<URI> ctlURIs = conService.getServiceDetails();
+        URI controllerRESTUri = ctlURIs.get(1);
+        Invocation.Builder builder;
+        Response response;
```

Comment:
The response should be closed too so that the underlying message entity input stream (if available and open) is closed and other resources associated with the response are released. You could perhaps use `@Cleanup` Lombok annotation for that.